include/arch/gdbstub.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_GDBSTUB_H_
+#define ZEPHYR_INCLUDE_ARCH_GDBSTUB_H_
+
+#include <inttypes.h>
+
+#if defined(CONFIG_X86)
+#include <arch/x86/ia32/gdbstub.h>
+#endif
+
+/**
+ * ARCH_GDB_NUM_REGISTERS is architecure specific and
+ * this symbol must be defined in architecure specific header
+ */
+#define GDB_NUM_REGISTERS ARCH_GDB_NUM_REGISTERS
+
+/**
+ * @brief Architecture layer debug start
+ *
+ * This function is called by @c z_gdb_start()
+ */
+void arch_gdb_start(void);
```

Comment:
these arch_ prototypes need to be moved to the arch interface headers. either kernel/include/kernel_arch_interface.h (preferred) or include/sys/arch_interface.h

Reply:
not sure about it, it is not kernel API.  `sys/arch_interface.h` says:
```
Any public kernel APIs that are implemented as inline functions and need to
call architecture-specific API so will have the prototypes for the
architecture-specific APIs here. 
```

Reply:
if I move these things to `kernel_arch_interface.h` I will need either move gdbstub code to kernel or include kernel internal headers to that subsystem.

Reply:
you have to put these in the arch interface headers.

Either in

`kernel/include/kernel_arch_interface.h` (preferred)
or
`include/sys/arch_interface.h`

You probably want the latter.

Reply:
can you be a little bit more specific ? are you proposing just move the API declaration to `include/sys/arch_interface.h` and keep the header `include/arch/gdbstub.h` that will be responsible to include architecture specific header `include/arch/x86/ia32/gdbtub.h` ? or get rid of `include/sys/arch_interface.h` and include `include/arch/x86/ia32/gdbstub.h` directly in `include/sys/arch_interface.h` ? I'm ok with the move, it is just looking a little bit an alien there though.